vectornav/src/main.cpp

```diff
@@ -461,3 +309,199 @@ void BinaryAsyncMessageReceived(void* userData, Packet& p, size_t index)
+                ROS_INFO("Connected baud rate is %d",vs.baudrate());
+                baudSet = true;
+            }
+        }
+        // Catch all oddities
+        catch(...){
+            // Disconnect if we had the wrong default and we were connected
+            vs.disconnect();
+            ros::Duration(0.2).sleep();
+        }
+        // Increment the default iterator
+        i++;
+        // There are only 9 available data rates, if no connection
+        // made yet possibly a hardware malfunction?
+        if(i > 8)
+        {
+            break;
+        }
+    }
+
+    // Now we verify connection (Should be good if we made it this far)
+    if(vs.verifySensorConnectivity())
+    {
+        ROS_INFO("Device connection established");
+    }else{
+        ROS_ERROR("No device communication");
+        ROS_WARN("Please input a valid baud rate. Valid are:");
+        ROS_WARN("9600, 19200, 38400, 57600, 115200, 128000, 230400, 460800, 921600");
+        ROS_WARN("With the test IMU 128000 did not work, all others worked fine.");
+    }
+    // Query the sensor's model number.
+    string mn = vs.readModelNumber();
+    ROS_INFO("Model Number: %s", mn.c_str());
+
+    // Set the device info for passing to the packet callback function
+    UserData user_data;
+    user_data.device_family = vs.determineDeviceFamily();
+
+    // Set Data output Freq [Hz]
+    vs.writeAsyncDataOutputFrequency(async_output_rate);
+
+    // Configure binary output message
+    BinaryOutputRegister bor(
+            ASYNCMODE_PORT1,
+            SensorImuRate / async_output_rate,  // update rate [ms]
+            COMMONGROUP_QUATERNION
+            | COMMONGROUP_ANGULARRATE
+            | COMMONGROUP_POSITION
+            | COMMONGROUP_ACCEL
+            | COMMONGROUP_MAGPRES,
+            TIMEGROUP_NONE,
+            IMUGROUP_NONE,
+            GPSGROUP_NONE,
+            ATTITUDEGROUP_YPRU, //<-- returning yaw pitch roll uncertainties
+            INSGROUP_INSSTATUS
+            | INSGROUP_POSLLA
+            | INSGROUP_POSECEF
+            | INSGROUP_VELBODY
+            | INSGROUP_ACCELECEF);
+
+    vs.writeBinaryOutput1(bor);
+
+    // Set Data output Freq [Hz]
+    vs.writeAsyncDataOutputFrequency(async_output_rate);
+    vs.registerAsyncPacketReceivedHandler(&user_data, BinaryAsyncMessageReceived);
+
+    // You spin me right round, baby
+    // Right round like a record, baby
+    // Right round round round
+    while (ros::ok())
+    {
+        diagnosticsUpdater.update();
+        ros::getGlobalCallbackQueue()->callAvailable(
+            ros::WallDuration(diagnosticsUpdater.getPeriod()));
+        // Need to make sure we disconnect properly. Check if all ok.
+    }
+
+    // Node has been terminated
+    vs.unregisterAsyncPacketReceivedHandler();
+    ros::Duration(0.5).sleep();
+    vs.disconnect();
+    ros::Duration(0.5).sleep();
+
+}
+
+int main (int argc, char *argv[]){
+    run(argc,argv);

```

Comment:
@patapufate did you mean 
```suggestion
    return run(argc,argv);
```
?

Reply:
fixed